### ISSUE TITLE
System function RegexMatch is not recognized in v4 preview3

### ISSUE THREAD

**Issue by Scythen** [Azure/azure-cosmos-dotnet-v3#2137]
Executing the following query results in the error **"'RegexMatch' is not a recognized built-in function name."**
`"SELECT * FROM c WHERE RegexMatch(c.attachment, 's...')"`

**To Reproduce**
Execute a query that uses the system function RegexMatch.

**Expected behavior**
The query should execute with no errors and return a set of records.

**Actual behavior**
Provide a description of the actual behavior observed. 

**Environment summary**
SDK Version: 4.0.0-preview3

**Additional context**
sample code that reproduces the error:
```
using Azure.Cosmos;
using Azure.Cosmos.Fluent;
using System;


var connectionString = "AccountEndpoint=https://localhost:8081/;AccountKey=C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==";
var cosmosClient = new CosmosClientBuilder(connectionString)
                    .WithConnectionModeGateway()
                    .Build();

var db = await cosmosClient.CreateDatabaseIfNotExistsAsync("Authorization");
var container = await db.Database.DefineContainer("Attachments", "/attachment").CreateIfNotExistsAsync();
var query = new QueryDefinition("SELECT * FROM c WHERE RegexMatch(c.attachment, 'b0d29ea1-ea94-435f-b4fa-da9305aa614d')");

await foreach(var it in container.Container.GetItemQueryIterator<Attachment>(query))
{
}

public class Attachment
{
    public Guid? id { get; set; }
    public string description { get; set; }
    public string attachment { get; set; }
}
```

**Comment by ealsur** [comment]
V4 does not have the required changes to support RegEx, are you doing some experiment and that is the reason you are using V4?

If this is an app that will need to go on production, please use V3.

**Comment by Scythen** [comment]
Yes, the app will eventually go to production but not for a while. It would be nice to continue using V4 as the API is significantly better.

What is the roadmap for v4?

**Comment by ealsur** [comment]
> Available in 6 - 12 months?
Not as far as I know.

Which API did you find better, the AsyncEnumerable support?

**Comment by Scythen** [comment]
Yes, AsyncEnumerable is one of the biggest improvements.
Also cleaning up naming CosmosDatabas vs Database for example.
The AsyncPageable also simplifies so much when implementing pagination in the API.
Better support for System.Text.Json too.
Lots of reasons to like V4. I may just use a UDF in place of the system function for now,

**Comment by Scythen** [comment]
Just looked at the V4 history and I don't see any activity, is this actually dead? If so, I will obviously have to switch to V3.
If it is dead, the documentation should be removed from the ms documentation site.

**Comment by ealsur** [comment]
Not dead, just not prioritized currently, focus currently is to stabilize V3

**Comment by Mortana89** [comment]
The inactivity is indeed painful. Last known date was GTM in May, 2020 that is. Because of the underlying work for the Azure.Core stuff.
It looks like it will take a while longer, well, we're all in IT, we know deadlines 😂
Our biggest reason to chose for V4 was the STJ support indeed, using our custom converters as well. We did this change when we went from 2.2 to 3.1.
We're looking forward to getting some of the new things on V4! (Batch, transactions, new SQL functions 💪💪)!

**Comment by Jeremywhiteley** [comment]
Another vote for v4
